File: src/types/magicalActions/_SkillReference.ts
import { SkillIdentifier } from "../_Identifier.js"

/**
 * A reference to a skill used for performing melodies.
 */
export type MusicalSkillReference = {
  /**
   * The skill's identifier.
   */
  id: SkillIdentifier<SkillIdentifierValue>
}

export type SkillIdentifierValue =
  | 9 // Singing
  | 56 // Music

File: Sources/OptolithDatabaseSchema/GeneratedTypes/magicalActions/_SkillReference.swift
//
//  _SkillReference.swift
//  OptolithDatabaseSchema
//

/// A reference to a skill used for performing melodies.
public struct MusicalSkillReference: EntitySubtype {
    /// The skill's identifier.
    public let id: SkillIdentifier

    public init(id: SkillIdentifier) {
        self.id = id
    }
}
